### PR TITLE
document partially visible indexes

### PR DESCRIPTION
**Pull request by taroface**
DOC-7501
DOC-7507
DOC-8003

- Document `VISIBILITY` clause for `ALTER INDEX`, `CREATE INDEX` (and implicitly `CREATE TABLE`, though that doc wasn't affected).
- Note that partially visible indexes are considered fully invisible for index recommendations.

This is covered in the 23.2, 24.1, and 24.2 docs.

Preview links containing the major changes: 
https://deploy-preview-18737--cockroachdb-docs.netlify.app/docs/stable/alter-index#visibility
https://deploy-preview-18737--cockroachdb-docs.netlify.app/docs/v24.1/cost-based-optimizer#control-whether-the-optimizer-uses-an-index

### CODE REVIEW

**Review by rytaft — APPROVED** [review]
LGTM thanks!

File: src/current/v23.2/create-index.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ Parameter | Description
...
+`opt_index_visible` | An optional `VISIBLE`, `NOT VISIBLE`, or `VISIBILITY` clause that indicates that an [index is visible, not visible, or partially visible to the cost-based optimizer]({% link {{ page.version.version }}/cost-based-optimizer.md %}#control-whether-the-optimizer-uses-an-index). If not visible, the index will not be used in queries unless it is specifically selected with an [index hint]({% link {{ page.version.version }}/indexes.md %}#selection) or the property is overridden with the [`optimizer_use_not_visible_indexes` session variable]({% link {{ page.version.version }}/set-vars.md %}#optimizer-use-not-visible-indexes). For examples, see [Set index visibility]({% link {{ page.version.version }}/alter-index.md %}#set-index-visibility).<br><br>Indexes that are not visible are still used to enforce `UNIQUE` and `FOREIGN KEY` [constraints]({% link {{ page.version.version }}/constraints.md %}). For more considerations, see [Index visibility considerations](alter-index.html#not-visible).
```
Comment (florence-crl) [inline src/current/v23.2/create-index.md:56]:
```suggestion
`opt_index_visible` | An optional `VISIBLE`, `NOT VISIBLE`, or `VISIBILITY` clause that indicates that an [index is visible, not visible, or partially visible to the cost-based optimizer]({% link {{ page.version.version }}/cost-based-optimizer.md %}#control-whether-the-optimizer-uses-an-index). If not visible, the index will not be used in queries unless it is specifically selected with an [index hint]({% link {{ page.version.version }}/indexes.md %}#selection) or the property is overridden with the [`optimizer_use_not_visible_indexes` session variable]({% link {{ page.version.version }}/set-vars.md %}#optimizer-use-not-visible-indexes). For examples, refer to [Set index visibility]({% link {{ page.version.version }}/alter-index.md %}#set-index-visibility).<br><br>Indexes that are not visible are still used to enforce `UNIQUE` and `FOREIGN KEY` [constraints]({% link {{ page.version.version }}/constraints.md %}). For more considerations, refer to [Index visibility considerations](alter-index.html#not-visible).
```

**Review by florence-crl — APPROVED** [review]
LGTM with non-blocking nits
- replacing `see` with `refer to`
- one grammar correction
- I know the section `Set an index to be not visible` example was not modified, but the results of `SHOW INDEXES FROM rides;` can be updated to include the `visibility` column. It mentions that `visible` is the last column, this is no longer correct since `visibility` is now the last column.

Please apply any correction to all versions.

**Comment by taroface** [comment]
>     * I know the section `Set an index to be not visible` example was not modified, but the results of `SHOW INDEXES FROM rides;` can be updated to include the `visibility` column. It mentions that `visible` is the last column, this is no longer correct since `visibility` is now the last column.

Done! Thanks for noticing this.

**Pull request merged**